build/Nuget/Microsoft.NET.Nuget.proj

```diff
@@ -25,29 +21,31 @@
   <Target Name="Build"
           DependsOnTargets="LogOutput;
                             InvalidateNuGetCache;
-                            @(NuSpec->'%(LayoutTargetName)')"
-          Inputs="@(NuSpec);@(PackageAssets)"
-          Outputs="@(NuSpec->'%(NuPkgOutput)')">
+                            @(NuSpec->'%(LayoutTargetName)')">
 
-    <PropertyGroup>
-      <NuGetExeVersion>3.4.4</NuGetExeVersion>
-      <NuGetTool>$(RepositoryRootDirectory).nuget\$(NuGetExeVersion)\nuget.exe</NuGetTool>
-      <NuGetDownloadUrl>https://dist.nuget.org/win-x86-commandline/v$(NuGetExeVersion)/NuGet.exe</NuGetDownloadUrl>
-      <NuGetDownloadCommand>powershell -noprofile -nologo -command "(new-object System.Net.WebClient).DownloadFile('$(NugetDownloadURL)', '$(NuGetTool)')"</NuGetDownloadCommand>
-    </PropertyGroup>
+    <ItemGroup>
+      <BinPlaceContent Include="$(MSBuildThisFileDirectory)Sdkproject.csproj" />
+      <BinPlaceContent Include="$(MSBuildThisFileDirectory)Microsoft.NET.Sdk.nuspec" />
+      <NuspecFile Include="$(InitialOutDir)Microsoft.NET.Sdk.nuspec" />
+    </ItemGroup>
 
+    <Copy 
+        SourceFiles="@(BinPlaceContent)"
+        DestinationFolder="$(InitialOutDir)"
+    />
+    <Exec Command="$(DotNetTool) restore $(InitialOutDir)" />
     <MakeDir Directories="$(NuGetOutDir)" />
+    <ReadLinesFromFile File="@(NuspecFile)" >
+         <Output TaskParameter="Lines" ItemName="ItemsFromFile" />
+    </ReadLinesFromFile>
+    <ItemGroup>
+        <NewLines Include="@(ItemsFromFile -> '$([System.String]::Copy('%(ItemsFromFile.Identity)').Replace('$version$','$(Version)'))' )" />
```

Comment:
Looking at the docs, it looks like you could pass `/p:NuspecProperties=\"Version=$(Version)\"` or something similar to set the version without having to modify the file.

It looks like you're updating the .nuspec file in-place, which means once you've run a build, the version number will be stamped in and won't get replaced by subsequent builds.  So I think that needs to be fixed one way or the other.